src/components/UserSelect/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import styles from './UserSelect.module.css';
+
+type UserSelectProps = {
+    user?: number,
+    idx: number,
+}
+
+function UserSelect(props: UserSelectProps) {
+    const dispatch = useDispatch();
+    const todos = useSelector((state: {list: { todos: any[] }}) => state.list.todos);
+    React.useEffect(
```

Comment:
you should extract fetching logic into hooks and AbortController should be used for cancelling requests when component unmounts or use React query